File: app/src/main/java/com/annapol04/munchkin/db/EventDao.java
package com.annapol04.munchkin.db;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import com.annapol04.munchkin.engine.Event;

import java.util.List;

@Dao
public interface EventDao {
    @Query("SELECT * from events")
    LiveData<List<Event>> loadEntries();

    @Insert
    void insert(Event event);
}

File: app/src/main/java/com/annapol04/munchkin/db/EventDao.kt
package com.annapol04.munchkin.db

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.Query

import com.annapol04.munchkin.engine.Event

@Dao
interface EventDao {
    @Query("SELECT * from events")
    fun loadEntries(): LiveData<List<Event>>

    @Insert
    fun insert(event: Event)
}
